bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -603,6 +609,59 @@ impl Room {
         let user_id = UserId::parse(&user_id)?;
         Ok(self.inner.get_suggested_user_role(&user_id).await?)
     }
+
+    pub fn default_power_levels(&self) -> RoomPowerLevels {
+        let levels = SdkRoom::default_power_levels();
+        RoomPowerLevels::from(levels)
+    }
+}
+
+#[derive(uniffi::Record)]
+pub struct RoomPowerLevels {
+    /// The level required to ban a user.
+    pub ban: i64,
+    /// The level required to invite a user.
+    pub invite: i64,
+    /// The level required to kick a user.
+    pub kick: i64,
+    /// The level required to redact an event.
+    pub redact: i64,
+    /// The default level required to send message events.
+    pub events_default: i64,
+    /// The default level required to send state events.
+    pub state_default: i64,
+    /// The default power level for every user in the room.
+    pub users_default: i64,
+    /// The level required to change the room's name.
+    pub room_name: i64,
+    /// The level required to change the room's avatar.
+    pub room_avatar: i64,
+    /// The level required to change the room's topic.
+    pub room_topic: i64,
+}
+
+impl From<RumaPowerLevels> for RoomPowerLevels {
+    fn from(value: RumaPowerLevels) -> Self {
+        let default_state: i64 = value.state_default.into();
+        let room_name =
+            value.events.get(&TimelineEventType::RoomName).map_or(default_state, |l| (*l).into());
```

Comment:
```suggestion
            value.events.get(&TimelineEventType::RoomName).map_or(default_state, |&lev| lev.into());
```

I'm not a fan of `l` as a variable name :-)

Also I _think_ I got it right that if you pattern-match `&lev` as the argument, then you don't need to `*` it to use it, but could be wrong.

Reply:
It should be handled a bit better in https://github.com/matrix-org/matrix-rust-sdk/pull/3183/commits/9e4db1a0d3b90cb3437ff19b97eb3d5d926f34dd